pkg/serverless/appsec/appsec.go

```diff
@@ -146,3 +157,9 @@ func wafHealth() error {
 	}
 	return nil
 }
+
+// canExtractSchemas checks that API Security is enabled
+// and that sampling rate allows schema extraction for a specific monitoring instance
+func (a *AppSec) canExtractSchemas() bool {
+	return a.cfg.APISec.Enabled && a.cfg.APISec.SampleRate >= rand.Float64()
```

Comment:
I'm not familiar with the math here: isn't it `a.cfg.APISec.SampleRate <= rand.Float64()` instead?

Reply:
No, here is an example (with what you wrote): 
if rate is 10% (0.1), then `rand.Float64()` will be greater or equal 9 times out of 10, resulting in an actual rate of 90% or 0.9.

Reply:
Another way to think about it: the higher the sample rate, the more chance it has to be >= to a random float number in the [0,1] space

Reply:
Thanks 🔢 ^^ All clear indeed